src/tests/ftest/util/job_manager_utils.py

```diff
@@ -1116,3 +1116,98 @@ def dump_logs(self, hosts=None, timestamp=None):
+
+        Returns:
+            str: the command with all the defined parameters
+
+        """
+        commands = [super().__str__(), "-w {}".format(self.hosts), str(self.job)]
+        return " ".join(commands)
+
+    def assign_hosts(self, hosts, path=None, slots=None):
+        """Assign the hosts to use with the command (--hostfile).
+
+        Args:
+            hosts (NodeSet): hosts to specify in the hostfile
+            path (str, optional): not used. Defaults to None.
+            slots (int, optional): not used. Defaults to None.
+        """
+        self._hosts = hosts.copy()
+
+    def assign_environment(self, env_vars, append=False):
+        """Assign or add environment variables to the command.
+
+        Args:
+            env_vars (EnvironmentVariables): the environment variables to use
+                assign or add to the command
+            append (bool): whether to assign (False) or append (True) the
+                specified environment variables
+        """
+        if append:
+            self.env.update(env_vars)
+        else:
+            self.set_environment(env_vars)
+
+    def run(self):
+        """Run the command.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        """
+        command = " ".join([self.env.to_export_str(), str(self.job)]).strip()
+        self.result = run_remote(self._hosts, command, self.verbose, self.timeout)
+
+        if self.result.timeout:
+            raise CommandFailure(
+                "Timeout detected running '{}' on {}".format(str(self.job), self.hosts))
+
+        if self.exit_status_exception and not self.check_results():
+            # Command failed if its output contains bad keywords
+            raise CommandFailure(
+                "Bad words detected in '{}' output on {}".format(str(self.job), self.hosts))
+
+        return self.result
+
+    def check_results(self):
+        """Check the command result for any bad keywords.
+
+        Returns:
+            bool: True if either there were no items from self.check_result_list to verify or if
+                none of the items were found in the command output; False if a item was found in
+                the command output.
+
+        """
+        status = True
+        if self.result and self.check_results_list:
+            regex = r"({})".format("|".join(self.check_results_list))
+            self.log.debug("Checking the command output for any bad keywords: %s", regex)
+            for data in self.result.output:
+                match = re.findall(regex, "\n".join(data.stdout))
+                if match:
+                    self.log.info(
+                        "The following error messages have been detected in the '%s' output:",
+                        str(self.job))
+                    for item in match:
+                        self.log.info("  %s", item)
+                    status = False
+                    break
+        return status
```

Comment:
Nitpick - I **think** this is the same
```suggestion
                    return False
        return True
```

Reply:
It may not.  If each host had slightly different output then the `self.result.output` would consist of more than one item.  The intent in this case would be to report all of the matching bad keywords on each set of hosts.  If we were to simply return, then the possibility may exist that a different bad keyword was seen on a different set of hosts and it would not be reported.